### ISSUE TITLE
Super High CPU usage.

### ISSUE THREAD

**Issue by vungsung** [AliasIO/Wappalyzer#2605]
Check out this link with Chrome enabled Wappalyzer:
https://notepad-plus-plus.org/community/topic/109/update-translation-on-github-repository

**Comment by kyletaylored** [comment]
I couldn't recreate this issue. Using Brave browser (Chrome) on macOS Mojave, Wappalyzer 5.7.4

**Comment by vungsung** [comment]
The bug is confirmed that still exists, even after 3 months and several Chrome updates...
Using Windows 10 / Chrome 73.0.3683.86 / Wappalyzer 5.7.4
Nothing showed in Developer Console. I only got a little information of threads in Process Hacker: `CPU usage:12%, Thread name: chrome.exe!GetHandleVerifier+0xbe670`
I also found that lots of threads were created and exited during the bug happens.
Thanks for your help.

**Comment by kyletaylored** [comment]
To try and limit the scope of this to Chrome, could you do the following?

1. Open "Manage Chrome Extensions"
2. Enable Developer mode (top right switch)
3. Click the link `html/background.html` under the new "Inspect Views" section (this open DevTools debug only for the Wappalyzer Chrome extension).

From there, you can run a clean profile.
1. Open the page: https://notepad-plus-plus.org/community/topic/109/update-translation-on-github-repository 
2. In the new DevTools, clear out any messages in the console tab (for clarity)
3. Go to the Performance tab, click "Record"
4. Go back to the Notepad++ page, click refresh.
5. While the page is loading, go back to the DevTools window, and when the page is completely finished, click stop.
6. Save the profile and upload it here.
7. Copy the console messages, paste here as well.

If anything, this should give some insight into what's going on. (**disclaimer, I'm not a maintainer, I've just been pretty active today 😄*). Below I've posted some of my results that I just ran:

<img width="1392" alt="wap-profile" src="https://user-images.githubusercontent.com/1759794/55916262-829a4c80-5bb1-11e9-8f32-6dba35777442.png">
<img width="1280" alt="wap-console" src="https://user-images.githubusercontent.com/1759794/55916263-8332e300-5bb1-11e9-802a-9d82dc96d309.png">

**Comment by vungsung** [comment]
Wow!! Thanks for all your effort!
Here are my performance records (Remember to remove the .txt extension): [Download](https://github.com/AliasIO/Wappalyzer/files/3066065/Profile-20190411T060744.json.txt)
And a screenshot of the console logs:
![console](https://user-images.githubusercontent.com/32069685/55916994-844f2900-5c20-11e9-9f9a-a7a2797d76b8.PNG)
Unfortunately, it seems no error was recorded by the console nor the performance recorder...
Thanks again for your help!

**Comment by kyletaylored** [comment]
Looking back at the profile, it looks like the total time spent by Wappalyzer is only 175ms, so I'm not sure this is necessarily a Wappalyzer issue, and potentially more of a Chrome issue (or something else is being spun at or around the same time?). I couldn't find anything that was obvious, other than maybe the garbage collector running?

Unless there's anymore info, I'd probably recommend closing this issue.

**Comment by vungsung** [comment]
Thanks for your remind. I'll comment once I find what cause it.

**Comment by vungsung** [comment]
Ok, I gathered a lot of information using Chrome's debugger during these 4 hours.
It's the `nodebb.js` of Notepad++ website in conflict with Wappalyzer's `iframe.js`.

Please take a look of these screenshots:
1) iframe.js (Before Runs Function):
![iframe js_beforerun](https://user-images.githubusercontent.com/32069685/60064176-2a93d080-9732-11e9-83eb-533ae923c2e5.png)
2) iframe.js (After Runs Function, it returns to nodebb.js):
![iframe js_return](https://user-images.githubusercontent.com/32069685/60064179-2a93d080-9732-11e9-8851-817b95635eeb.png)
3) nodebb.js (Before Runs Function):
![nodebb js_beforerun](https://user-images.githubusercontent.com/32069685/60064175-29fb3a00-9732-11e9-8400-994009e5fffc.png)
4) nodebb.js (After Runs Function, it returns to iframe.js, the first screenshot):
![nodebb js_return](https://user-images.githubusercontent.com/32069685/60064178-2a93d080-9732-11e9-9a24-712f9a33c86f.png)

It takes infinite loop. I don't get it why, I googled "nodebb.js infinite loop", but there's not much cases. IDK it's the Wappalyzer's bug or NP++'s outdated nodebb.js. Thanks for your time on reading.

**Comment by kyletaylored** [comment]
This is the craziest thing. I can't for the life of me get Wappalyzer to render the iframe.js script, even though I see it listed in the manifest for the Chrome extension, so I can't recreate that issue.
 
![image](https://user-images.githubusercontent.com/1759794/60068564-e8887180-96d4-11e9-90e2-27b74e116ca4.png)

But what it seems like is that both scripts are listening to the PostMessage API, which is a system for sending messages through iFrames. So Wappalyzer is listening for a message and parsing it, then the nodebb.js script is periodically sending a new message, and Wappalyzer keeps processing them. 

What I don't know is, why is the iframe.js script loading? It seems like it might be specific to ads? Not sure, but that seems like the right path.

**Comment by vungsung** [comment]
It's so great to hear there's new progress on this! And thanks for your explanation of what these API does. So the next problem: Is iframe.js loaded in a specific condition, and how. I'll try to find it out, thanks again.

**Comment by vungsung** [comment]
I've got some bad news: This might be due to a bug in Google Chrome.
After removing Wappalyzer, extension Grammarly inherited the problem, lopping message event with nodebb.js.

Recently I'm dealing with lots of iframe and frame bugs in my Google Chrome with my projects. It would be something like "not fired elements". One of the cases is: https://stackoverflow.com/questions/38894306/javascript-returns-null-in-chrome-html-element-in-source

Those elements in frames(or iframes) would be null until users inspect them in developer tool.

This happens in my friend and my Android phone.
All Chronium-based browser won't be able to browse a specific website,
until I use Android Chromium's remote debugging tool, "remote inspecting" the element in my computer.

And then at the day before yesterday, I was building a project, to use events to exploit Chrome's bug. And then, jquery was stuck fetching an element until I inspect it in developer tool.

After all of these, and then the "iframe.js", makes me think it's all of Chrome's bug.
Today, I ask my friend to help me do the steps: 1. Install Wappalyzer 2. Open the problem NP++ page. Expected, nothing happened.

Maybe there are some corrupted files in my Google Chrome installed directory? NO. I've tried clean installed the whole Windows 10. But the extensions, I enabled Google Chrome Syncing, which means all the extensions would be reinstalled back once I login to my account.

So, there's a big possibility that one or more extensions "conflicted" <-(so unprofessional I know, but that's it, I totally have no idea what happens inside the Chrome core / The Nuclear Reactor.) and cause the bug.

Thanks for all your help, if I notice something later I'll let you know first :)
